src/components/App.jsx

```diff
@@ -0,0 +1,143 @@
+import React, { Component } from 'react';
+import './App.css';
+import FlightList from './FlightList';
+
+export default class App extends Component {
+  constructor(props) {
+    super(props);
+    this.data = [];
+
+    this.state = {
+      statusDate: 'today',
+      statusFlight: 'departures',
+
+      yesterdayDepartures: '',
+      yesterdayArrival: '',
+      todayDepartures: '',
+      todayArrival: '',
+      tomorrowDepartures: '',
+      tomorrowArrival: ''
+    };
+    this.clickHandler = this.clickHandler.bind(this);
+  }
+
+  parseData(dataArrs) {
+    this.setState(() => ({
+      yesterdayDepartures: dataArrs[0].body.departure,
+      yesterdayArrival: dataArrs[0].body.arrival,
+      todayDepartures: dataArrs[1].body.departure,
+      todayArrival: dataArrs[1].body.arrival,
+      tomorrowDepartures: dataArrs[2].body.departure,
+      tomorrowArrival: dataArrs[2].body.arrival
+    }));
+  }
+
+  createData() {
+    const date = new Date();
+    const url = this.props.urlFlight;
+
+    const dateYesterday = `${date.getDate()}-${date.getMonth() + 1}-${date.getFullYear()}`;
+    const dateToday = `${date.getDate() + 1}-${date.getMonth() + 1}-${date.getFullYear()}`;
+    const dateTomorrow = `${date.getDate() + 2}-${date.getMonth() + 1}-${date.getFullYear()}`;
+
+    const dataYesterday = url + dateYesterday;
+    const dataToday = url + dateToday;
+    const dataTomorrow = url + dateTomorrow;
+
+    this.data.push(dataYesterday, dataToday, dataTomorrow);
+  }
+
+  componentDidMount() {
+    this.createData();
+    this.init();
+  }
+
+  createPromise(url) {
+    return fetch(url).then(resp => resp.json());
+  }
+
+  getAllData(urls) {
+    return Promise.all(urls.map(url => this.createPromise(url)));
```

Comment:
It's bad practice to download all data together.
Download by default only `today` all other only by the click on the button.